refactor(components): refactor steps/step

emojiiii opened the pull request:
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.
#3113

emojiiii committed:
refactor(components): refactor steps/step

Alanscut committed:
refactor(components): refactor progress

emojiiii committed:
refactor(components): change validator to values

undefined-degisn commented:
i dont' know why the test often make mistakes

emojiiii committed:
refactor(components): refactor steps/step

sxzz commented on packages/components/steps/src/steps.ts:
```diff
@@ -0,0 +1,48 @@
+    default: 'process',
+    values: ['wait', 'process', 'finish', 'error', 'success'],
+  },
+})
```
```suggestion
} as const)
```

sxzz commented on packages/components/steps/src/step.vue:
```diff
@@ -60,29 +60,10 @@ import {
-  props: IStepsProps
-  steps: Ref<StepItemState[]>
-}
+import { elStepsKey } from '@element-plus/tokens/steps'
```
I suggest to use `stepsContextKey`

sxzz commented on packages/components/steps/src/step.vue:
```diff
@@ -91,32 +72,16 @@ export default defineComponent({
     const internalStatus = ref('')
-    const parent: IStepsInject = inject('ElSteps')
-    const currentInstance = getCurrentInstance()
+    const parent: StepsInject = inject(elStepsKey)!
+
+    if (!parent) throwError('ElStep', 'must be nested inside ElSteps')
```
```suggestion
    const parent = inject(elStepsKey)
    if (!parent) throwError(COMPONENT_NAME, 'must be nested inside ElSteps')
```

sxzz replied:
Also, please move `inject` and `getCurrentInstance` to first line.

sxzz commented on packages/components/steps/src/step.ts:
```diff
@@ -0,0 +1,40 @@
+  calcProgress: (status: string) => void
+}
+
+export interface StepsInject {
```
move this type to `tokens` and rename to `StepsContext`

sxzz commented on packages/tokens/steps.ts:
```diff
@@ -0,0 +1,4 @@
+import type { StepsInject } from '@element-plus/components'
+import type { InjectionKey } from 'vue'
+
+export const elStepsKey: InjectionKey<StepsInject> = Symbol('ElSteps')
```
```suggestion
export const stepsContextKey: InjectionKey<StepsContext> = Symbol('stepsContextKey')
```

emojiiii committed:
refactor(components): refactor steps/step

sxzz commented on packages/components/steps/src/step.vue:
```diff
@@ -60,63 +60,30 @@ import {
 import { Close, Check } from '@element-plus/icons'
 
-import type { Ref, PropType, Component } from 'vue'
+import { stepsContextKey } from '@element-plus/tokens/steps'
```
```suggestion
import { stepsContextKey } from '@element-plus/tokens'
```

sxzz commented on packages/tokens/steps.ts:
```diff
@@ -0,0 +1,5 @@
+import type { StepsContext } from '@element-plus/components'
```
please move `StepsContext` to here

sxzz commented on packages/components/steps/src/step.vue:
```diff
@@ -60,63 +60,30 @@ import {
-  },
+  props: stepProps,
   setup(props) {
+    const parent: StepsContext = inject(stepsContextKey)!
```
- We can remove annotation since stepsContextKey has its type.
- Also, please remove `!`

sxzz commented on packages/components/steps/src/step.vue:
```diff
@@ -60,63 +60,30 @@ import {
-    const parent: IStepsInject = inject('ElSteps')
-    const currentInstance = getCurrentInstance()
+
+    const currentInstance = getCurrentInstance()!
```
Please move to Line 79

sxzz reviewed (commented):
It seems that some reviews you haven't changed it.

emojiiii committed:
refactor(components): refactor steps/step

YunYouJun committed:
refactor(style): rename to text-color bg-color (#4468)

* refactor(style): rename -font to -text-color & -background to -bg-color

* refactor(style): rename bg-color text-color

* refactor(style): rename 'color' to 'text-color'

YunYouJun committed:
fix(style): radio checkbox button default color (#4492)

* refactor(style): adjust component size to large/default/small

* fix(style): radio checkbox button default color

* Revert "refactor(style): adjust component size to large/default/small"

This reverts commit a952d9c81a6370816133ef4543e2c948fa3706f2.

* fix(style): use disabled-color-base as button disabled-text-color

YunYouJun committed:
refactor(style): rename disabled color and migrate to css var (#4493)

* refactor(style): adjust component size to large/default/small

* fix(style): radio checkbox button default color

* Revert "refactor(style): adjust component size to large/default/small"

This reverts commit a952d9c81a6370816133ef4543e2c948fa3706f2.

* fix(style): use disabled-color-base as button disabled-text-color

* refactor(style): rename disabled base color

* refactor(style): simplify css var generate

kooriookami committed:
fix: coler picker support touch (#4477)

* fix: coler picker support touch

* fix: update

* fix: add getClientXY util

* fix: move getClientXY to util/dom

renovate[bot] committed:
chore(deps): update dependency @vueuse/core to v7.0.1 (#4494)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency rollup to v2.60.1 (#4496)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update commitlint monorepo to v15 (major) (#4437)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency lint-staged to v12 (#4391)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency @vitejs/plugin-vue to v1.10.0 (#4497)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

emojiiii committed:
refactor(components): refactor button (#4499)

emojiiii committed:
fix(components): [ElFormItem] label will be show undefined (#4487)

muhammadcahya committed:
fix(locale): improve Indonesian (Bahasa) translations and fix typo (#4476)

renovate[bot] committed:
chore(deps): update dependency ts-morph to v13.0.1 (#4500)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update pnpm to v6.23.1 (#4501)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency ts-morph to v13.0.2 (#4503)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency vite-plugin-inspect to v0.3.11 (#4511)

renovate[bot] committed:
chore(deps): update dependency @vueuse/core to v7.0.3 (#4510)

sxzz committed:
refactor: improve tsconfig (#4386)

oliverzy committed:
perf(components): avoid unnecessary tab switch rerender (#4513)

Co-authored-by: 周毓杰 <xiake@xiaohongshu.com>

xiaoxian521 committed:
fix(components): [el-menu] menu-collapse onLeave lose done (#4517)

sumy7 committed:
fix(components): [el-upload] fix get accept from uploader.props (#4506)

fix #4505

renovate[bot] committed:
chore(deps): update dependency @pnpm/find-workspace-packages to v3.1.25 (#4520)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update pnpm to v6.23.2 (#4521)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency @vueuse/core to v7.1.0 (#4526)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

emojiiii committed:
fix(components): [date-picker] `range-separator slot` don't work (#4522)

renovate[bot] committed:
chore(deps): update dependency sass to v1.43.5 (#4527)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency esbuild to v0.14.0 (#4551)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency @vueuse/core to v7.1.2 (#4529)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update jamesives/github-pages-deploy-action action to v4.1.7 (#4548)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency @vitejs/plugin-vue to v1.10.1 (#4542)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency @types/sass to v1.43.1 (#4534)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

element-bot committed:
chore: update changelog 1.2.0-beta.4 (#4553)

renovate[bot] committed:
chore(deps): update dependency eslint-define-config to v1.2.0 (#4538)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

yonghui-wang committed:
Update item.vue (#4555)

remove useless code

renovate[bot] committed:
chore(deps): update dependency prettier to v2.5.0 (#4536)

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: Kevin <sxzz@sxzz.moe>

renovate[bot] committed:
chore(deps): update dependency @pnpm/find-workspace-packages to v3.1.26 (#4559)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

YunYouJun committed:
fix(components): confirm button style when in message-box (#4564)

sxzz committed:
refactor(hooks): refactor hooks (#4253)

* refactor(hooks): remove use-css-var

* refactor(hooks): remove use-events

* refactor(hooks): remove use-migrating

* refactor(hooks): remove use-transition

* refactor(hooks): named export useAttrs

* refactor(hooks): named export useFocus

* refactor(hooks): refactor useFormItem

* refactor(hooks): refactor useGlobalConfig

* refactor(hooks): refactor useLocale

* refactor(hooks): refactor useLockscreen

* refactor(hooks): refactor useModal

* refactor(hooks): refactor useModelToggle

* refactor(hooks): refactor usePreventGlobal

* refactor(hooks): refactor useRestoreActive

* refactor(hooks): refactor useTeleport

* refactor(hooks): refactor useThrottleRender

* refactor(hooks): refactor useTimeout

* refactor(hooks): refactor useTransitionFallthrogh

sxzz committed:
feat(docs): add auto import and update icon (#4554)

YunYouJun committed:
fix(style): input inner padding right when has suffix (#4574)

emojiiii committed:
fix(components): [ElMenu] the collapsible menu switch will disappear (#4575)

0song committed:
chore: update doc sidebar (#4561)

btea committed:
improvement(components): [el-cascader-panel] unref store (#4544)

emojiiii committed:
fix(components): [ElScrollbar] fix useResizeObserver watch undefined (#4576)

Alanscut committed:
fix(components): [el-tag] icon-center (#4438)

Alanscut committed:
fix(components): [el-dropdown] icon vertical center (#4431)

renovate[bot] committed:
chore(deps): update dependency rollup to v2.60.2 (#4586)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency vue-tsc to v0.29.8 (#4562)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency esbuild to v0.14.1 (#4585)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update pnpm to v6.23.3 (#4597)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency sass to v1.44.0 (#4583)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update typescript-eslint monorepo to v5.5.0 (#4581)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

sxzz committed:
chore(deps): update dependency vue to 3.2.23 (#4599)

sxzz committed:
fix(utils): setTimeout & config-provider type (#4598)

* fix(utils): setTimeout type

* fix(components): export config-provider types

element-bot committed:
chore: update changelog 1.2.0-beta.5 (#4610)

EstebanFuentealba committed:
fix(theme-chalk): add !optional to @extend

emojiiii committed:
feat(components): [ElCalender] add internationalization (#4582)

* feat(components): [ElCalender] add internationalization

* perf(components): [ElCalender] remove redundant  `ref`

Alanscut committed:
fix(components): [el-date-picker] can't use numpadEnter (#4563)

* fix(components): [el-date-picker] can't use numpadEnter

* fix(components): [el-date-picker]enter to close when input.value is null

* fix(components): [el-date-picker] add test

emojiiii committed:
fix(hooks): rename `ConfigProvdierContext` to `ConfigProviderContext` (#4620)

renovate[bot] committed:
chore(deps): update pnpm to v6.23.4 (#4624)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency @pnpm/find-workspace-packages to v3.1.27 (#4623)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

YunYouJun committed:
fix(components): dropdown menu divided style, close #4631 (#4638)

Alanscut committed:
fix(components): [el-config-provider] autoInsertSpace false (#4629)

oscaralbareda committed:
fix(locale): update ca (#4644)

cokemine committed:
chore(play): fix dev environment (#4643)

renovate[bot] committed:
chore(deps): update dependency @pnpm/find-workspace-packages to v3.1.28 (#4647)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update pnpm to v6.23.5 (#4648)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

btea committed:
fix(components): [el-button] computed type (#4641)

YunYouJun committed:
fix(style): icon font-size inherit (#4565)

msidolphin committed:
docs: [el-tree-v2] fixed the description of props.value (#4376)

renovate[bot] committed:
chore(deps): update pnpm to v6.23.6 (#4659)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

sxzz committed:
fix(hooks): fix translate option (#4619)

sxzz committed:
refactor: change icon package name (#4625)

renovate[bot] committed:
chore(deps): update dependency unplugin-vue-components to v0.17.3 (#4663)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency esbuild to v0.14.2 (#4661)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

btea committed:
fix(components): [el-message] fit badge error type (#4662)

renovate[bot] committed:
chore(deps): update dependency eslint to v8.4.0 (#4660)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency prettier to v2.5.1 (#4666)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

Alanscut committed:
refactor(components): refactor drawer (#4282)

Co-authored-by: Kevin <sxzz@sxzz.moe>

renovate[bot] committed:
chore(deps): update dependency unplugin-vue-components to v0.17.4 (#4667)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

sxzz committed:
fix(hooks): return key name if not defined (#4664)

sxzz committed:
chore(deps): remove unless dependencies (#4668)

0song committed:
chore: update doc nav new (#4669)

YunYouJun committed:
refactor(style): rename -color-base to --el-bg-color (#4615)

YunYouJun committed:
fix: upgrade @element-plus/icons-vue to currentColor (#4671)

renovate[bot] committed:
chore(deps): update dependency eslint-plugin-vue to v8.2.0 (#4688)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency unplugin-vue-components to v0.17.6 (#4675)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update typescript-eslint monorepo to v5.6.0 (#4695)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

sxzz committed:
fix: vue type errors (#4696)

* chore(deps): upgrade vue to v3.2.24

* fix(deps): add csstype

zhangshichun committed:
fix(components): [el-table] tooltip wrong style

fix #4697

Alanscut committed:
fix(components): [el-input-number]raise native form validation error (#4706)

KawaiiZapic committed:
fix(hooks): [lockscreen] body lock remove unexpected (#4693) (#4694)

Co-authored-by: KawaiiZapic <kawaiizapic@zapic.moe>

imguolao committed:
fix(docs): fix code not working in Codepen (#4678) (#4679)

fix #4678

spx443812507 committed:
feat(components): [el-dialog] support custom close icon (#4514)

[el-dialog] support custom close icon

Alanscut committed:
fix(components): [el-input-number]set value equal to min will update (#4650)

renovate[bot] committed:
chore(deps): update dependency @vueuse/core to v7.2.2 (#4727)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency eslint to v8.4.1 (#4702)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency vite to v2.7.1 (#4710)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

fanhefeng committed:
docs: update quickstart.md (#4723)

pdsuwwz committed:
fix(docs): replace @element-plus/icons with @element-plus/icons-vue (#4704)

renovate[bot] committed:
chore(deps): update dependency rollup to v2.61.0 (#4742)

renovate[bot] committed:
chore(deps): update dependency @vitejs/plugin-vue to v1.10.2 (#4709)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

Commit:
fix(docs): fix webpack config import syntax (#4718)

renovate[bot] committed:
chore(deps): update dependency @types/lodash to v4.14.178 (#4732)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

Alanscut committed:
feat(components): [el-divider] support setting the dividing line style  (#4435)

* feat(components): [el-divider] support setting the dividing line style

* Update divider.md

YunYouJun committed:
fix(style): override el-icon display in el-input__icon (#4684)

* fix(style): override el-icon display in el-input__icon

* fix(style): use double class selector to override input icon style

* Update packages/theme-chalk/src/input.scss

Co-authored-by: Aex <adaex@qq.com>

Alanscut committed:
docs: [el-date-picker] add example of how to use prefix-icon (#4736)

Alanscut committed:
feat(components): [el-select] tag type (#4504)

Alanscut committed:
refactor(components): refactor calendar (#4325)

Co-authored-by: Kevin <sxzz@sxzz.moe>

sxzz committed:
refactor(components): refactor progress

renovate[bot] committed:
chore(deps): update dependency markdown-it to v12.3.0 (#4749)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency components-helper to v1.0.5 (#4767)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency typescript to v4.5.3 (#4751)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

sxzz committed:
feat(components): [el-config-provider] add size (#4730)

sxzz committed:
refactor: improve types (#4746)

sxzz committed:
refactor(components): refactor loading (#4750)

Alanscut committed:
fix(components): [el-menu] third-party icons are hidden (#4754)

iceshou committed:
refactor(components): [el-checkbox] simplify the value (#4674)

Co-authored-by: iceman <dd>

CarterLi committed:
fix(components): [table-column, time-picker] improve typings (#4761)

sxzz committed:
fix(build): compatible to build on windows (#4747)

emojiiii committed:
refactor(components): refactor switch (#4271)

Alanscut committed:
docs: [el-cascader] add guidance on how to set props (#4689)

Alanscut committed:
fix(components): [el-tooltip] fallback placement (#4771)

element-bot committed:
chore: Update changelog 1.2.0-beta.6 (#4775)

emojiiii committed:
fix(components): [ElDatePicker] click clear is invalid (#4768)

sxzz committed:
ci: add issue helper (#4779)

sxzz committed:
fix(build): copy README (#4782)

sxzz committed:
ci: refactor pr preview alert (#4781)

msidolphin committed:
fix: [el-date-picker] focus input box when focus method is called (#4343)

fix #4327

btea committed:
fix(components): [date-picker] filter custom content comment (#4651)

iamkun committed:
chore: update readme

iamkun committed:
chore: update readme

iamkun committed:
chore: update readme

iamkun committed:
chore: update preview img (#4788)

renovate[bot] committed:
chore(deps): update actions-cool/maintain-one-comment action to v2 (#4784)

renovate[bot] committed:
chore(deps): update dependency rollup to v2.61.1 (#4789)

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: Kevin <sxzz@sxzz.moe>

renovate[bot] committed:
chore(deps): update dependency unplugin-vue-components to v0.17.8 (#4776)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency sass to v1.45.0 (#4785)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

sxzz committed:
fix(build): circular dependency (#4791)

YunYouJun committed:
refactor(utils): use @ctrl/tinycolor instead of custom color.ts (#4682)

YunYouJun committed:
refactor(style): adjust component size to large/default/small (#4491)

* refactor(style): adjust component size to large/default/small

* refactor(components): avatar size & use flex instead of block

* refactor(components): adjust check button size

* refactor(components): adjust tag size

* refactor(components): adjust size doc

* fix(components): datetime-picker demo style width

* refactor(components): color-picker size & block to flex

* refactor(components): adjust slider input size

* refactor(components): adjust radio input size for demo

* refactor(components): adjust select size & docs

* refactor(components): adjust form radio size & docs

* refactor(components): add windicss for docs

* refactor(components): adjust skeleton avatar size to css var

* refactor(components): simplify typography size demo

* refactor(components): adjust dropdown size & demo

* refactor(components): adjust descriptions size

* fix(components): datetime-picker showcase class pollute global button

* chore(ci): upgrade docs dependencies to fix ci

* fix(ci): add highlight because vitepress not export it

* fix(ci): disable line for no-console

* fix(ci): remove mini to fix test

* fix(style): code font size

* fix(style): button span flex style

* fix(style): button padding horizontal default 15px

* refactor(components): adjust tag padding size & demo

* refactor(components): adjust form line-height for input

* refactor(components): adjust dropdown menu size & button padding

* fix(style): picker separator block to flex center

* fix: dropdown button span items-center

* style: adjust input-with-icon & size demo & fix input vitepress load

* chore: upgrade dependencies

* chore: upgrade dependencies

* ci: fix website build

* ci: regenerate pnpm-lock.yaml

* ci: use dev pnpm-lock

* ci: update pnpm-lock.yaml

YunYouJun committed:
fix(style): custom icon size (#4798)

renovate[bot] committed:
chore(deps): update dependency @vitejs/plugin-vue to v2 (#4797)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): pin dependencies (#4799)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

sxzz committed:
refactor(utils): remove isServer (#4792)

sxzz committed:
chore: update issue workflow (#4794)

YunYouJun committed:
feat(style): add scrollbar for docs (#4800)

* feat(style): add scrollbar for docs

* fix(style): adjust scrollbar color

btea committed:
chore(utils): remove unnecessary as string (#4805)

renovate[bot] committed:
chore(deps): update dependency esbuild to v0.14.3 (#4802)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency unplugin-vue-components to v0.17.9 (#4812)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency @rollup/plugin-node-resolve to v13.1.1 (#4814)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency vite to v2.7.2 (#4811)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update typescript-eslint monorepo to v5.7.0 (#4816)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

cregis0073 committed:
fix(theme-chalk): Put back !optional to @extend (#4835)

Alanscut committed:
fix(components): [el-form] vertical aligin (#4717)

YunYouJun committed:
fix(style): adjust component size & demo (#4801)

* fix(style): adjust button size & demo spacing

* fix(style): adjust input padding & with prefix/suffix

* fix(style): adjust button padding horizontal

* fix(style): adjust form margin & font-size

* refactor(docs): use setup simplify form examples

YunYouJun committed:
refactor(docs): use setup sugar simplify input demo (#4845)

YunYouJun committed:
refactor(components): set messagebox button size be default (#4846)

YunYouJun committed:
chore: move \@ctrl/tinycolor to dependencies (#4842)

renovate[bot] committed:
chore(deps): update pnpm to v6.24.1 (#4832)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency @pnpm/find-workspace-packages to v3.1.30 (#4827)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency @crowdin/cli to v3.7.3 (#4831)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency @vitejs/plugin-vue to v2.0.1 (#4828)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency typescript to v4.5.4 (#4818)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency @pnpm/types to v7.7.1 (#4829)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency esbuild to v0.14.5 (#4865)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency @crowdin/cli to v3.7.4 (#4881)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

fanhefeng committed:
docs: [el-time-picker] update default value (#4898)

already disabledTime '18:40', it might mislead reader.

fanhefeng committed:
docs: [el-form] typo (#4897)

Alanscut committed:
docs: [el-upload]add example of setting on-exceed to cover the old file (#4861)

* docs: [el-upload]add example of seting on-exceed to cover the old file

* Update packages/components/upload/src/useHandlers.ts

Co-authored-by: btea <2356281422@qq.com>

* Update docs/examples/upload/limit-cover.vue

Co-authored-by: btea <2356281422@qq.com>

Co-authored-by: btea <2356281422@qq.com>

thadguidry committed:
docs: fix grammar (#4893)

thadguidry committed:
docs: improve grammar (#4894)

thadguidry committed:
docs: improve grammar (#4895)

Commit:
feat(components): [el-dropdown] add new "popper-class" prop (#4909)

Co-authored-by: João Gonçalves <joaogoncalves@pixelmatters.com>

yin1999 committed:
docs: update alibaba npm mirror registry (#4911)

xiaoxian521 committed:
perf: [el-switch] activeIcon and inactiveIcon type (#4899)

imswk committed:
fix(components): [el-popper] onVisibilityChange (#4907)

msidolphin committed:
fix: [el-checkbox-group]: default value adjusted to empty array (#4316)

fix #4300

imswk committed:
chore(docs): modify tooltip base style (#4900)

* chore(docs): modify tooltip base style

YunYouJun committed:
fix: tag in select style (#4914)

* fix: tag in select style

* fix: extract common style from select to fix select

emojiiii committed:
docs(components): use `script setup` replace `defineComponent` (#4849)

* docs(components): use `script setup` replace `defineComponent`

* docs(components): use setup sugar simplify `checkbox` demo

* docs(components): use setup sugar simplify `collapse` demo

* docs(components): use setup sugar simplify `color-picker` demo

* docs(components): use setup sugar simplify `config-provider` demo

* docs(components): use setup sugar simplify `container` demo

* docs(components): use setup sugar simplify `date-picker` demo

* docs(components): use setup sugar simplify `datetime-picker` demo

* docs(components): use setup sugar simplify `dialog` demo

* docs(components): use setup sugar simplify `drawer` demo

* docs(components): use setup sugar simplify `dropdown` demo

* docs(components): use setup sugar simplify `image` demo

* docs(components): use setup sugar simplify `infinite-scroll` demo

* docs(components): use setup sugar simplify `input-number` demo

* docs(components): use setup sugar simplify `loading` demo

* docs(components): use setup sugar simplify `menu` demo

* docs(components): use setup sugar simplify `message` demo

* docs(components): use setup sugar simplify `message-box` demo

* docs(components): use setup sugar simplify `notification` demo

* docs(components): use setup sugar simplify `page-header` demo

* docs(components): use setup sugar simplify `pagination` demo

* docs(components): use setup sugar simplify `popover` demo

* docs(components): use setup sugar simplify `progress` demo

* docs(components): use setup sugar simplify `radio` demo

* docs(components): use setup sugar simplify `rate` demo

* docs(components): use setup sugar simplify `scrollbar` demo

* docs(components): use setup sugar simplify `select` demo

* docs(components): use setup sugar simplify `select-v2` demo

* docs(components): use setup sugar simplify `skeleton` demo

* docs(components): use setup sugar simplify `slider` demo

* docs(components): use setup sugar simplify `space` demo

* docs(components): use setup sugar simplify `step` demo

* docs(components): use setup sugar simplify `switch` demo

* docs(components): use setup sugar simplify `tabs` demo

* docs(components): use setup sugar simplify `tag` demo

* docs(components): use setup sugar simplify `time-picker` demo

* docs(components): use setup sugar simplify `time-select` demo

* docs(components): use setup sugar simplify `timeline` demo

* docs(components): use setup sugar simplify `tooltip` demo

* docs(components): use setup sugar simplify `transfer` demo

* docs(components): use setup sugar simplify `transition` demo

* docs(components): use setup sugar simplify `tree` demo

* docs(components): use setup sugar simplify `table` demo

* docs(components): Escape character

* docs(components): use setup sugar simplify `tree-v2` demo

* docs(components): use setup sugar simplify `upload` demo

* docs(components): use `ref` to replace `reactive`

* docs(ci): use `playground` to replace `codepen`

* docs(ci): change icon from `codepen`  to `ep`

* docs(components): `sass` to `css`

* fix(components): add button CssVar

* docs(ci): change `Edit in Codepen.io` to `Edit in Playground`

* docs(ci): add vitepress i18

HerringtonDarkholme committed:
test(components): remove several sleep from test code (#4924)

btea committed:
docs: add edit-in-editor tooltip (#4918)

Ryan2128 committed:
fix(components): [select-v2] missing validate action when clear (#4892)

* fix(components): [select-v2] missing validate action when clear

- Fix the bug that missing validate when clear
- Add the same icon and style as the select component

* docs: [select-v2] fix demo render error

renovate[bot] committed:
chore(deps): update dependency vite to v2.7.6 (#4867)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

HerringtonDarkholme committed:
test(components): remove more sleep from input test (#4937)

HerringtonDarkholme committed:
test(components): fix time picker default value in cancel test (#4938)

Alanscut committed:
feat: [el-popconfirm]add debounce of confirm() and cancel() (#4943)

renovate[bot] committed:
chore(deps): update dependency @pnpm/find-workspace-packages to v3.1.31 (#4931)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency lint-staged to v12.1.3 (#4887)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency @vueuse/core to v7.4.1 (#4796)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency eslint to v8.5.0 (#4883)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency lint-staged to v12.1.4 (#4956)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency pretty-quick to v3.1.3 (#4957)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update jamesives/github-pages-deploy-action action to v4.1.8 (#4888)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update pnpm to v6.24.2 (#4961)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency esbuild to v0.14.8 (#4955)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency sass to v1.45.1 (#4959)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update typescript-eslint monorepo to v5.8.0 (#4966)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency vite-plugin-windicss to v1.6.1 (#4964)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

HerringtonDarkholme committed:
fix(core): fix unit testing warning (#4948)

HerringtonDarkholme committed:
fix(components): change size mini to small (#4947)

HerringtonDarkholme committed:
fix(components): always emit scroll with scrollTop number (#4974)

HerringtonDarkholme committed:
fix(components): fix more size prop checks (#4975)

renovate[bot] committed:
chore(deps): update dependency vite to v2.7.7 (#4976)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update pnpm to v6.24.3 (#4973)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency unplugin-vue-components to v0.17.11 (#4960)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency rollup-plugin-esbuild to v4.8.1 (#4980)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency vitepress to v0.20.10 (#4101)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update commitlint monorepo to v16 (major) (#4981)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

HerringtonDarkholme committed:
fix(components): fix message warning (#4979)

purepear committed:
feat(components): [date/time-picker] add id attribute to the inputs (#4972)

* feat(components): [date/time-picker] add id attribute to the inputs

* docs(components): [date/time-picker] add id to attributes

cn-troy committed:
fix(components): [el-step] 移除在simple模式下多余的点 (#4944)

Alanscut committed:
docs: [el-upload]add description of handleRemove() (#4983)

renovate[bot] committed:
chore(deps): update dependency rollup to v2.62.0 (#4962)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

Commit:
fix(components): [el-loading] fix the overlap problem

* fix(components): [el-loading] fix overlap  bug

sxzz committed:
docs: change playground domain (#4989)

YunYouJun committed:
fix: message-box confirm button style (#4988)

renovate[bot] committed:
chore(deps): update dependency @typescript-eslint/eslint-plugin to v5.8.1 (#4993)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

emojiiii committed:
feat(components): [ElCheckboxGroup] add Fragment (#4743)

msidolphin committed:
fix(components): [el-drawer] fix content cannot scroll issue (#5001)

fix #4999

renovate[bot] committed:
chore(deps): update dependency @typescript-eslint/parser to v5.8.1 (#4994)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

satrong committed:
docs: [el-input] add password type (#5002)

Co-authored-by: Kevin <sxzz@sxzz.moe>

renovate[bot] committed:
chore(deps): update dependency @commitlint/cli to v16.0.1 (#5004)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

fanhefeng committed:
fix(components): [el-upload] improve image style (#4902)

ITKHMER committed:
fix(locale): update translation/spelling of Khmer (#4930)

Co-authored-by: Kevin <sxzz@sxzz.moe>

sxzz committed:
refactor: upgrade vue & remove `@vue/compiler-sfc` (#4815)

* refactor: remove @vue/compiler-sfc

* chore(deps): upgrade vue

YunYouJun committed:
fix(components): input-number/input/dropdown/select/badge details (#4866)

thadguidry committed:
docs: improve grammar (#4896)

Co-authored-by: 三咲智子 <sxzz@sxzz.moe>

renovate[bot] committed:
chore(deps): update dependency vue-tsc to v0.30.1 (#4965)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency eslint-define-config to v1.2.1 (#5010)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency vite to v2.7.8 (#5007)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

Alanscut committed:
fix(components): [el-input-number]set undefined to modelValue (#4869)

YunYouJun committed:
fix(theme-chalk): upload progress & menu padding style & pagination align (#5011)

* fix: upload progress margin

* fix: menu example style

* fix: picker & radio padding, button form details

* fix(theme-chalk): pagination align

renovate[bot] committed:
chore(deps): update dependency @pnpm/types to v7.8.0 (#5032)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency rollup-plugin-esbuild to v4.8.2 (#5029)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency vite to v2.7.9 (#5016)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency @rollup/plugin-replace to v3.0.1 (#5017)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency esbuild to v0.14.9 (#5019)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update pnpm to v6.24.4 (#5039)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency @pnpm/find-workspace-packages to v3.1.32 (#5031)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

emojiiii committed:
refactor(components): refactor steps/step

emojiiii committed:
refactor(components): refactor steps/step

emojiiii committed:
Merge branch 'refactor-steps' of https://github.com/undefined-degisn/element-plus into refactor-steps
